openedx/core/lib/xblock_builtin/xblock_discussion/xblock_discussion.py

```diff
@@ -88,6 +94,49 @@ def django_user(self):
             return None
         return user_service._django_user  # pylint: disable=protected-access
 
+    @staticmethod
+    def vendor_js_dependencies():
+        """
+        Returns list of vendor JS files that this XBlock depends on.
+        """
+        return get_js_dependencies('discussion_vendor')
+
+    @staticmethod
+    def js_dependencies():
+        """
+        Returns list of JS files that this XBlock depends on.
+        """
+        return get_js_dependencies('discussion')
+
+    @staticmethod
+    def css_dependencies():
+        """
+        Returns list of CSS files that this XBlock depends on.
+        """
+        if get_language_bidi():
+            return ['css/inline-discussion-rtl.css']
+        else:
+            return ['css/inline-discussion.css']
+
+    def add_resource_urls(self, fragment):
+        """
+        Adds URLs for JS and CSS resources that this XBlock depends on to `fragment`.
+        """
+        # Head dependencies
+        for vendor_js_file in self.vendor_js_dependencies():
+            fragment.add_resource_url(static(vendor_js_file), "application/javascript", "head")
+
+        for css_file in self.css_dependencies():
+            fragment.add_css_url(static(css_file))
+
+        # Body dependencies
+        fragment.add_javascript(loader.load_unicode("static/js/mathjax_include.js"))
```

Comment:
@itsjeyd sorry that wasn't clear! I was referring to the files that all get bundled into discussion.js via this rule:

https://github.com/edx/edx-platform/blob/master/lms/envs/common.py#L1575

What I want to ensure is that in production we don't include extra JavaScript in the rendered HTML for the block, but at the same time we don't introduce an extra file download. The ideal situation to my mind is that everything goes into discussion.js.